integrations/docker/images/chip-build/Dockerfile

```diff
@@ -53,6 +53,9 @@ RUN set -x \
     udev \
     ninja-build \
     strace \
+    avahi-daemon \
```

Comment:
is the daemon a *build* dependency?

Reply:
We use the chip-build docker to run unit tests as well and the mdns test requires avahi-daemon to be running.

Reply:
I believe this is a prerequisite for #3241 - on linux our mDNS implementation uses avahi.

Reply:
Do these tests send real advertisements out on the network?

Reply:
Yes, they send real mdns packets.

Reply:
The basic test suite shouldn't be sending out advertisements onto the local network. It's surprising, pollutes the network, and it may cause tests to be flaky depending on the network environment. Communicating via localhost or on an isolated network (e.g. like the happy tests) would be fine. Can we do it here?

I guess installing avahi-daemon might be OK if we condition these tests on a flag (I dunno, "enable_network_tests" ?) since docker containers should already be on an isolated network.

See also a related discussion on #2644.

Reply:
Approved with the expectation that we will keep these tests separate from those that don't require any particular network access.

It's a bit of a shame to require `avahi-daemon` it even for jobs that build firmwares.. perhaps we should move test running into a different docker image.

Reply:
Moved the mdns test to happy tests so that avahi-daemon can be installed in the happy bootstrap script.